$OpenBSD: patch-src_libtracker-miner_tracker-monitor_c,v 1.3 2012/01/01 09:02:11 ajacoutot Exp $

Recognize the gio-kqueue monitoring backend.
(not to be pushed upstream before the glib changes are there first)

--- src/libtracker-miner/tracker-monitor.c.orig	Fri Dec 30 15:12:53 2011
+++ src/libtracker-miner/tracker-monitor.c	Fri Dec 30 15:17:59 2011
@@ -294,6 +294,15 @@ tracker_monitor_init (TrackerMonitor *object)
 			 */
 			priv->monitor_limit = MAX (priv->monitor_limit, 0);
 		}
+		else if (strcmp (name, "GKqueueDirectoryMonitor") == 0) {
+			/* Using kqueue(2) */
+			g_message ("Monitor backend is kqueue(2)");
+
+			/*
+			 * XXX use getrlimit(2) (or its glib equivalent)
+			 */
+			priv->monitor_limit = 400;
+		}
 		else if (strcmp (name, "GFamDirectoryMonitor") == 0) {
 			/* Using Fam */
 			g_message ("Monitor backend is Fam");
